### PR TITLE
Feat(bigquery): allow first part of table names to contain dashes

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Reference: https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#table_names

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -2245,6 +2247,16 @@ def _parse_table_part() -> t.Optional[exp.Expression]:
             )
 
         table = _parse_table_part()
+
+        # https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#table_names
+        if self.TABLE_NAME_ALLOWS_DASHES and isinstance(table, exp.Identifier):
```
Comment (tobymao) [inline sqlglot/parser.py:2252]: why is this done here, shouldn't this be in parse_table_parts?

Comment (GeorgeSittas) [reply sqlglot/parser.py:2252]: Dashes are only allowed to the leftmost identifier according to the documentation, so I did this here to 1) capture that and 2) reduce unneeded overhead when parsing the rest of the parts.

Comment (GeorgeSittas) [reply sqlglot/parser.py:2252]: If this seems like unnecessary visual noise to have in the base parser, I can factor it out and have it only in BigQuery.

Comment (tobymao) [reply sqlglot/parser.py:2252]: let's just put this in bigquery because it's so unstandard

Comment (GeorgeSittas) [reply sqlglot/parser.py]: done

File: sqlglot/dialects/bigquery.py [hunk 2]
```diff
@@ -224,6 +224,24 @@ class Parser(parser.Parser):
             "OPTIONS": lambda self: exp.Properties(expressions=self._parse_with_property()),
         }
 
+        def _parse_table_part(self, schema: bool = False) -> t.Optional[exp.Expression]:
+            this = (
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:228]: can you make this a first class function and call it in both places so it's not copy and pasted?

Comment (GeorgeSittas) [reply sqlglot/dialects/bigquery.py:228]: sure

File: sqlglot/dialects/bigquery.py [hunk 3]
```diff
@@ -224,6 +224,24 @@ class Parser(parser.Parser):
...
+                (not schema and self._parse_function())
+                or self._parse_id_var(any_token=False)
+                or self._parse_string_as_identifier()
+            )
+
+            # https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#table_names
+            if isinstance(this, exp.Identifier):
+                table_name = this.name
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:236]: what if it's a-b-c?

Comment (GeorgeSittas) [reply sqlglot/dialects/bigquery.py:236]: It's handled in the following loop:

```python
>>> import sqlglot
>>> sqlglot.transpile("select * from a-b-c", "bigquery")
['SELECT * FROM a-b-c']
```

File: sqlglot/dialects/bigquery.py (hunk 2 above)
Comment (GeorgeSittas) [reply sqlglot/dialects/bigquery.py]: done

File: tests/dialects/test_bigquery.py [hunk 4]
```diff
@@ -6,6 +6,10 @@ class TestBigQuery(Validator):
     dialect = "bigquery"
 
     def test_bigquery(self):
+        self.validate_identity("SELECT * FROM a-b-c")
```
Comment (myschkyna) [inline tests/dialects/test_bigquery.py:9]: Can we also add an example with multiple hyphens in combination with tokens of type TokenType.DOT?

```sql
SELECT * FROM a-b-c.mydataset.mytable
```

**Review by tobymao — APPROVED** [review]

File: tests/dialects/test_bigquery.py (hunk 4 above)
Comment (GeorgeSittas) [reply tests/dialects/test_bigquery.py:11]: done

**Pull request merged**